packages/manager/modules/exchange/src/account/account.service.js

```diff
@@ -89,6 +91,141 @@ export default class ExchangeAccount {
     });
   }
 
+  /**
+   * @param {string} serviceName
+   * @param {object} userPrincipalName
+   */
+  createMfa(serviceName, userPrincipalName) {
+    return this.OvhApiMsServices.Account()
+      .Mfa()
+      .v6()
+      .create(
+        {
+          serviceName,
+          userPrincipalName,
+        },
+        {},
+      )
+      .$promise.then((data) => {
+        this.Exchange.refreshViews('Accounts', 'Tasks');
```

Comment:
I suppose there is no other way around ? The service shouldn't be triggering a view refresh

Reply:
Not yet, `exchange` isn't "full-`uirouter`" yet, and it's currently done like this.

Reply:
Damn, who rewrote this section two years ago and didn't change that ???